From 80925010473b8ea558863263fca784b96e5fc880 Mon Sep 17 00:00:00 2001
From: shkosugi <sin.kosugi@gmail.com>
Date: Sat, 26 Sep 2026 21:44:29 +0900
Subject: [PATCH 7/7] System.Web: forward IHtmlString to System.Web.HttpUtility

.NET has System.Web.IHtmlString in System.Web.HttpUtility; .NET Framework libraries built
against System.Web (System.Web.WebPages.Razor, MVC) failed to load it (TypeLoadException in
the WebPageRazorHost type initializer). The only public type of the .NET Framework System.Web
the port neither defines nor forwards, apart from RegiisUtility (IIS registration).
---
 src/WebFormsForCore.Web/IHtmlString.cs | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/WebFormsForCore.Web/IHtmlString.cs b/src/WebFormsForCore.Web/IHtmlString.cs
index 2efd895..82b446e 100644
--- a/src/WebFormsForCore.Web/IHtmlString.cs
+++ b/src/WebFormsForCore.Web/IHtmlString.cs
@@ -13,4 +13,9 @@ namespace System.Web {
         string ToHtmlString();
     }
 }
-#endif
\ No newline at end of file
+#endif
+#if !NETFRAMEWORK
+// On .NET the interface is in System.Web.HttpUtility. .NET Framework libraries built against System.Web
+// (System.Web.WebPages, MVC) look for it here.
+[assembly: System.Runtime.CompilerServices.TypeForwardedTo(typeof(System.Web.IHtmlString))]
+#endif
-- 
2.55.0.windows.3

